src/main/java/com/redhat/cloud/notifications/routers/OApiService.java

```diff
@@ -23,20 +21,22 @@
     @GET
     @Path("/{what}/v1.0/openapi.json")
     @Produces(MediaType.APPLICATION_JSON)
-    public StreamingOutput serveOpenAPI(@PathParam("what") String what) {
+    public Uni<String> serveOpenAPI(@PathParam("what") String what) {
 
         String resourceName = "/openapi." + what + ".json";
-        URL target = getClass().getResource(resourceName);
-        if (target == null) {
-            throw new WebApplicationException(resourceName, NOT_FOUND);
-        }
 
-        return new StreamingOutput() {
-            @Override
-            public void write(OutputStream output) throws IOException, WebApplicationException {
-                InputStream is = getClass().getResourceAsStream(resourceName);
-                is.transferTo(output);
-            }
-        };
+        return Uni.createFrom().item(resourceName)
+                .onItem().transform(n ->
+                        getClass().getResourceAsStream(n)
+                )
+                .onItem().ifNull().failWith(new WebApplicationException(resourceName, NOT_FOUND))
+                .onItem().transform(j -> {
+                    try {
+                        return j.readAllBytes();
```

Comment:
Novel effort, but this is still blocking code. The simplest way is probably to use vertx, like:

```java
import io.vertx.mutiny.core.Vertx;

@Inject
Vertx vertx;

return vertx.fileSystem().readFile(resourceName).onItem().transform(b -> b.toString("UTF-8"));
```

There are probably more sane ways to do it also with vertx - but these files are relatively small.